File: movie-ticket-system/java-without-multithreading/entities/Seat.java
package entities;
import enums.SeatType;
import enums.SeatStatus;

public class Seat {
    private String id;
    private int row;
    private int column;
    private SeatType type;
    private double price;
    private SeatStatus status;

    public Seat(String id, int row, int column, SeatType type, double price, SeatStatus status) {
        this.id = id;
        this.row = row;
        this.column = column;
        this.type = type;
        this.price = price;
        this.status = status;
    }

    public String getId() { return id; }
    public double getPrice() { return price; }
    public SeatStatus getStatus() { return status; }
    public void setStatus(SeatStatus status) { this.status = status; }
}

File: movie-ticket-system/go-without-multithreading/entities/seat.go
package entities

import "app/movie-ticket-system/go-without-multithreading/enums"

// Seat captures explicit graph positions inside a specific show theater mapping.
type Seat struct {
	ID     string
	Row    int
	Column int
	Type   enums.SeatType
	Price  float64
	Status enums.SeatStatus
}
